src/core/cuda_common_cuda.cu

```diff
@@ -336,26 +326,45 @@ void copy_part_data_to_gpu(ParticleRange particles) {
   }
 }
 
+std::unique_ptr<PinnedVectorHost<float>> particle_forces_host{
```

Comment:
I really don't like this construction. I a way this is worse than manual memory management, because it pretends to be RAII, but really isn't. As far as I can see there is not reason for these buffers to exist statically, couldn't you just create them e.g. when the integration starts, and release them after. Depending on the cost of the allocation, it may even be feasible to create it in the force calculation.